src/mongo_types.ts

```diff
@@ -499,19 +499,21 @@ export type PropertyType<Type, Property extends string> = string extends Propert
  * returns tuple of strings (keys to be joined on '.') that represent every path into a schema
  * https://docs.mongodb.com/manual/tutorial/query-embedded-documents/
  */
-export type NestedPaths<Type> = Type extends
-  | string
-  | number
-  | boolean
-  | Date
-  | RegExp
-  | Buffer
-  | Uint8Array
-  | ((...args: any[]) => any)
-  | { _bsontype: string }
+export type NestedPaths<Type, Depth extends number[]> = Depth['length'] extends 10
```

Comment:
1) Let's make sure we document why we're using whatever depth number we go with both here and in our public facing recursive type limitation docs

2) We may be able to pull in additional test examples from NODE-4594, or at least check to see how well this solution addresses the performance concerns raised there